src/components/legend.ts

```diff
@@ -280,6 +282,7 @@ export module Components {
       entries.select("path").attr("d", (d: any, i: number, j: number) => this.symbol()(d, j)(layout.textHeight * 0.6))
                             .attr("transform", "translate(" + (layout.textHeight / 2) + "," + layout.textHeight / 2 + ")")
                             .attr("fill", (value: string) => this._colorScale.scale(value))
+                            .attr("opacity", (d: any, i: number) => this.symbolOpacity()(d, i))
```

Comment:
Haha, good catch -- I wrote a unit test on the indices and it had the same issue. I've updated it to do the same thing as symbol, passing in d and j instead of d and i.